packages/@ember/-internals/container/lib/registry.ts

```diff
@@ -415,7 +415,7 @@ export default class Registry implements IRegistry {
     this._options[normalizedName] = options;
   }
 
-  getOptions(fullName: string): TypeOptions {
+  getOptions(fullName: string): TypeOptions | undefined {
```

Comment:
If the type wasn't registered and there was no fallback, this would be undefined.